benchmarks/dataset_serialize_benchmark.py

```diff
@@ -0,0 +1,232 @@
+import itertools
+import logging
+import os
+import shutil
+import time
+import uuid
+
+import conbench.runner
+import pyarrow
+import pyarrow.dataset as ds
+
+from benchmarks import _benchmark
+
+log = logging.getLogger(__name__)
+
+
+@conbench.runner.register_benchmark
+class DatasetSerializeBenchmark(_benchmark.Benchmark):
+    """
+    This benchmark is supposed to measure the time it takes to write data from
+    memory (from a pyarrow Table) to a tmpfs file system, given a specific
+    serialization format (parquet, arrow, ...).
+
+    To make this benchmark agnostic to disk read performance on the input side
+    of things, the data is read fully into memory before starting (and timing)
+    the benchmark function. That is (believed to be) achieved with:
+
+        data = source_dataset.to_table(filter=somefilter, ...)
```

Comment:
If we are performing this outside of the measurements, then the filter is also not being measured, right? I'm all for keeping the code, though, because I think it would make sense to eventually expand this benchmark into measuring an end to end read - filter - write workload.

Reply:
> If we are performing this outside of the measurements, then the filter is also not being measured, right? 

Right!

This following benchmark times read-from-disk-backed-fs-and-then-filter: https://github.com/voltrondata-labs/benchmarks/blob/5ea34d76951be9a323683344c5233310eb867908/benchmarks/dataset_selectivity_benchmark.py#L71

And this one does, too:

https://github.com/voltrondata-labs/benchmarks/blob/5ea34d76951be9a323683344c5233310eb867908/benchmarks/dataset_filter_benchmark.py#L31

Although I would think that in both cases the filtering does not dominate the time, but instead it's the network-attached disk (see below).

For focusing a benchmark on the filter performance, the data should be in memory already, using e.g. tmpfs (`/dev/shm`), like I am proposing here.

> I think it would make sense to eventually expand this benchmark into measuring an end to end read - filter - write workload.

Yeah, interesting thought. Maybe we want this kind of test/benchmark in the future. As described in #123 in fact here I started with that. And then I wondered what the use is. I called this "RDFSW":

`reading -> deserializing -> filtering -> serializing -> writing`

and I found that the signal on the final two stages was weak / non-existant, because of the read-from-network-attached-disk on the left-hand side of the flow. This part is both slow and volatile compared to the other steps. 

If we do this, we should read from tmpfs (which limits dataset size, of course).

That's a great discussion for subsequent work!